File: ChessApp.Server/Models/GameDetails.cs
namespace ChessApp.Server.Models
{
    public class GameDetails
    {
        public required float TimeLeft { get; set; }
        public required bool CanEnPassant { get; set; }
        public required bool CanCastleKingSide { get; set; }
        public required bool CanCastleQueenSide { get; set; }

    }
}

File: chessApp.client/src/types/GameDetails.ts


export type GameDetails = {
    timeLeft: number;
    canEnPassant: boolean; //true if player moved pawn two squares ahead and enemy pawn was nearby
    canCastleKingSide: boolean; //false if king or rook (on king side) moved
    canCastleQueenSide: boolean; //false if king or rook (on queen side) moved
}